File: sdk/node/src/bus.ts
import { connect, NatsConnection } from "nats";

/** NATS connection configuration. */
export interface NATSConfig {
  url: string;
  token?: string;
  user?: string;
  pass?: string;
  /** Client name reported to the NATS server. Defaults to `"cap-sdk-node"`. */
  name?: string;
}

/**
 * Opens a NATS connection using the provided configuration.
 *
 * @param cfg - Connection settings.
 * @returns A connected NATS client.
 */
export async function connectNATS(cfg: NATSConfig): Promise<NatsConnection> {
  return connect({
    servers: cfg.url,
    token: cfg.token,
    user: cfg.user,
    pass: cfg.pass,
    name: cfg.name ?? "cap-sdk-node",
  });
}

File: sdk/ruby/lib/cordum/cap/bus.rb
# frozen_string_literal: true

require 'nats/io/client'

module Cordum
  module Cap
    module Bus
      # Connects to a NATS server and returns the client.
      #
      # @param url [String] NATS server URL (e.g. "nats://localhost:4222")
      # @param token [String, nil] optional auth token
      # @param name [String, nil] optional client name
      # @return [NATS::IO::Client]
      def self.connect(url:, token: nil, name: nil)
        nc = NATS::IO::Client.new
        opts = { servers: [url] }
        opts[:token] = token if token
        opts[:name] = name if name
        nc.connect(opts)
        nc
      end
    end
  end
end
